ContentPatcher/docs/i18n/zh/extensibility.md

```diff
@@ -0,0 +1,303 @@
+
+## 目录
+* [入门](#introduction)
+* [访问API](#access-the-api)
+* [基本API](#basic-api)
+  * [概念](#concepts)
+  * [添加令牌](#add-a-token)
+* [进阶API](#advanced-api)
+  * [注意事项](#caveats)
+  * [概念](#concepts-1)
+  * [添加令牌](#add-a-token-1)
+* [参见](#see-also)
+
+## 入门<a name="introduction"></a>
+
+你的SMAPI模组可以使用Content Patcher的[模组API](https://zh.stardewvalleywiki.com/模组:制作指南/APIs/Integrations#模组API)来添加自定义令牌。自定义令牌的前缀为提供它们的模组，如`your-mod-id/SomeTokenName`。
+
+你可以用这两种API：
+
+* 大部分模组推荐使用 **基本API**。你可以在不了解Content Patcher内部结构的情况下创建令牌；Content Patcher会自动管理各种细节，而且你的令牌和未来版本更兼容。
+
+* **进阶API**提供更多控制选项。但是你的令牌会更加复杂，你需要了解Content Patcher的内在运行逻辑，而且你的令牌可能会在未来Content Patcher内在运行逻辑更改时失效。进阶API强烈不推荐使用，除非基本API不可用。
+
+你可以同时使用基本和进阶API。
+
+## 访问API<a name="access-the-api"></a>
+
+访问API的步骤为：
+
+1. 将Content Patcher设为[`manifest.json`中的依赖](https://stardewvalleywiki.com/Modding:Modder_Guide/APIs/Manifest#Dependencies):
+
+   ```js
+   "Dependencies": [
+      { "UniqueID": "Pathoschild.ContentPatcher", "IsRequired": false }
+   ]
+   ```
+
+2. 把[`IContentPatcherAPI`](../../../IContentPatcherAPI.cs) 复制到你模组里并删除**任何你不需要用的方法，为了兼容未来更改**.
+3. 在你的模组代码中（如[`GameLaunched`事件](https://zh.stardewvalleywiki.com/模组:制作指南/APIs/Events#GameLoop.GameLaunched)中，获取Content Patcher的API：
+   ```c#
+   var api = this.Helper.ModRegistry.GetApi<IContentPatcherAPI>("Pathoschild.ContentPatcher");
+   ```
+4. 使用API延伸Content Patcher功能，（详见以下）。
+
+## 基本API<a name="basic-api"></a>
+### 概念<a name="concepts"></a>
+
+基本API会替你处理大部分设计因素。你只需要考虑以下两点：
+
+<dl>
+<dt>范围</dt>
+<dd>
+
+Content Patcher只有在更新令牌时才会调用你的代码。这可以在存档加载前，加载中，和加载后。如果你的令牌还没准备好，你可以返回null或空列表 这三种情况在 _[添加令牌](#add-a-token)_ 中都照顾到。
+
+</dd>
+
+<dt>数据顺序</dt>
+<dd>
+
+你返回的顺序会影响`valueAt`。推荐使用对于你令牌来说最有意义的顺序，因为内容包作者无法改变此顺序。大部分令牌使用字母数字顺序即可（如 `.OrderBy(p => p, StringComparer.OrdinalIgnoreCase)`）。
+
+</dd>
+</dl>
+
+### 添加令牌<a name="add-a-token"></a>
+
+你可以在`GameLaunched`事件中通过`RegisterToken`添加简单令牌（详见 _[访问API](#access-the-api)_ ）。例如，以下代码创建一个`{{your-mod-id/PlayerName}}`令牌代表当前玩家名：
+
+```c#
+api.RegisterToken(this.ModManifest, "PlayerName", () =>
+{
+    // 存档已加载
+    if (Context.IsWorldReady)
+        return new[] { Game1.player.Name };
+
+    // 存档正在加载
+    if (SaveGame.loaded?.player != null)
+        return new[] { SaveGame.loaded.player.Name };
+
+    // 未加载存档(e.g. 在主页)
+    return null;
+});
+```
+
+`RegisterToken` 有三个参数：
+
+参数   | 类型 | 用途
+---------- | ---- | -------
+`mod`      | `IManifest` | 提供此令牌模组的manifest，你可以直接使用ModEntry的`this.ModManifest`
+`name`     | `string` | 令牌名称，这个名字只需要在你自己模组中独一。Content Patcher会自动给这个名字添加前缀。以上例子的`PlayerName`会变成`your-mod-id/PlayerName`。
+`getValue` | `Func<IEnumerable<string>>` | 一个函数，反馈当前值。如果这反馈null或空列表，这个令牌会以不存在于当前范围来处理，任何使用它的补丁和动态令牌也会被禁用。
+
+现在，任何将你的模组列为依赖的内容包都可以在其字段中使用令牌：
+```js
+{
+   "Format": "2.7.0",
+   "Changes": [
+      {
+         "Action": "EditData",
+         "Target": "Characters/Dialogue/Abigail",
+         "Entries": {
+            "Mon": "Oh hey {{your-mod-id/PlayerName}}! Taking a break from work?"
```

Comment:
We should probably translate this example text too (and the one at the bottom of this page).